.github/workflows/test_build_release_edge.yaml

```diff
@@ -51,14 +51,20 @@ jobs:
     runs-on: "ubuntu-latest"
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
       - uses: actions/setup-python@v2
         with:
           python-version: '3.x'
           architecture: 'x64'
-      - name: "Install charmcraft and build charms"
+      - name: "Install charmcraft"
         run: |
           pip install charmcraft
```

Comment:
As a good practice, I usually run `pip install -U pip` first. What's your thoughts on implementing here?

Reply:
Is it necessary? I thought using the version supplied with the `setup-python` action was enough. What's the advantage of updating pip here?
The downside is that this CI already takes a long time to run (~20 min or so), adding more steps will increase this time.

Reply:
I think it is better to always be deterministic about the versions, even the pip version